File: atcoder/abc244B_go_straight_and_turn_right.py
# uninhm
# https://atcoder.jp/contests/abc244/tasks/abc244_b
# implementation

n = int(input())
t = input()

x, y = 0, 0

facing = 'E'
cardinals = ['E', 'S', 'W', 'N']
movement = {
    'E': (1, 0),
    'S': (0, -1),
    'W': (-1, 0),
    'N': (0, 1)
}
for i in t:
    if i == 'S':
        mov = movement[facing]
        x += mov[0]
        y += mov[1]
    elif i == 'R':
        facing = cardinals[(cardinals.index(facing) + 1) % 4]
    else:
        raise ValueError

print(x, y)

File: atcoder/abc244B_go_straight_and_turn_right.nim
# vicfred
# https://atcoder.jp/contests/abc244/tasks/abc244_b
# implementation
import sequtils, strutils

let n = stdin.readLine.parseInt
let s = stdin.readLine

var x, y : int

var direction : int
for ch in s:
  if ch == 'R':
    direction += 1
    direction = direction mod 4
    continue
  case direction:
    of 0:
      x += 1
    of 1:
      y -= 1
    of 2:
      x -= 1
    of 3:
      y += 1
    else:
      echo "piola"
echo x, " ", y

